lib/smart_proxy_salt/rest.rb

```diff
@@ -0,0 +1,31 @@
+require 'smart_proxy_salt/api_request'
+require 'json'
+
+module Proxy::Salt::Rest
+  extend ::Proxy::Log
+  extend ::Proxy::Util
+
+  class << self
+    def environments_list
+      Proxy::Salt::ApiRequest.new.post('/run', :fun => 'fileserver.envs', :client => 'runner')['return'][0].to_s
+    end
+
+    def states_list(environment)
+      states = []
+      files = Proxy::Salt::ApiRequest.new.post('/run', :fun => 'fileserver.file_list',
+                                                       :saltenv => environment,
+                                                       :client => 'runner')['return'][0]
+
+      files.each do |file|
+        if file =~ /\.sls\Z/
+          states << file.gsub('init.sls', '')
+                        .gsub('.sls', '')
+                        .chomp('/')
+                        .gsub('/', '.')
+        end
+      end
```

Comment:
Given the following directory structure:
```
.
├── apache
│   ├── bar.sls
│   ├── baz
│   │   ├── asdf.sls
│   │   └── qux.sls
│   ├── foo.sls
│   └── init.sls
└── core.sls
```

This returns the following list of states:
```
["apache.bar", "apache.baz.asdf", "apache.baz.qux", "apache.foo", "apache", "core"]
```
I probably should improve the processing a be a bit more resilient... but  @bechtoldt Does the result seem right?